src/thread.cr

```diff
@@ -63,8 +68,10 @@ class Thread
 
       current_thread_id = LibC.pthread_self
 
-      current_thread = @@threads.find do |thread|
-        LibC.pthread_equal(thread.id, current_thread_id) != 0
+      current_thread = @@threads.each_key do |thread|
+        if LibC.pthread_equal(thread.id, current_thread_id) != 0
+          break thread
+        end
```

Comment:
I have to overload `Hash#hash_key`, cause this file runs before `StdHasher @@seed` initialized. In previous version I overloaded `Thread#hash(hasher)` to use unseeded permutation.

Which way is better? I'll bind to whatever you prefer.